File: backend/app/models/agent.py
from sqlalchemy import Column, String, Integer, Boolean, JSON, DateTime, ForeignKey, Float
from sqlalchemy.sql import func
from app.db.database import Base

class AgentModel(Base):
    __tablename__ = "agents"

    id             = Column(String, primary_key=True)
    employee_id    = Column(Integer, unique=True, nullable=False)
    name           = Column(String, nullable=False)
    role           = Column(String, nullable=False)
    tier           = Column(String, nullable=False)          # founder/executive/manager/worker
    department     = Column(String, nullable=False)
    model          = Column(String, nullable=False)
    personality_note = Column(String, default="")
    skills         = Column(JSON, default=[])
    status         = Column(String, default="idle")          # idle/thinking/working/chatting/training/terminated
    hired_at       = Column(DateTime, server_default=func.now())
    strikes        = Column(JSON, default=[])
    rewards        = Column(JSON, default=[])
    manager_id     = Column(String, ForeignKey("agents.id"), nullable=True)
    direct_reports = Column(JSON, default=[])
    memory_context = Column(String, default="")
    is_founder     = Column(Boolean, default=False)
    tile_x         = Column(Integer, default=0)
    tile_y         = Column(Integer, default=0)
    is_active      = Column(Boolean, default=True)
    quality_score  = Column(Float, default=100.0)
    created_at     = Column(DateTime, server_default=func.now())

File: frontend/src/types/agent.ts
export type AgentStatus = 'idle' | 'thinking' | 'working' | 'chatting' | 'training' | 'terminated'
export type AgentTier = 'founder' | 'executive' | 'manager' | 'worker'
export type Department = 'executive' | 'engineering' | 'research' | 'writing' | 'legal' | 'ld'
export type AIModel = 'claude-3-5-sonnet' | 'gpt-4o' | 'gemini-1.5-pro'
export type SkillLevel = 'junior' | 'mid' | 'senior' | 'expert'

export interface AgentSkill {
  name: string
  level: SkillLevel
}

export interface Strike {
  id: string
  reason: string
  severity: 'minor' | 'major'
  date: string
  resolved: boolean
  trainingCompleted: boolean
}

export interface Reward {
  id: string
  type: 'badge' | 'promotion' | 'commendation'
  title: string
  date: string
  reason: string
}

export interface Agent {
  id: string
  employeeId: number
  name: string
  role: string
  tier: AgentTier
  department: Department
  model: AIModel
  personalityNote: string
  skills: AgentSkill[]
  status: AgentStatus
  hiredAt: string
  strikes: Strike[]
  rewards: Reward[]
  managerId: string | null
  directReports: string[]
  memoryContext: string
  isFounder: boolean
  tilePosition: { x: number; y: number }
}
